activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -316,7 +316,7 @@ def exists?(conditions = :none)
         MSG
       end
 
-      return false if !conditions
+      return false if conditions.blank?
```

Comment:
I can see this backfiring on `Hash#blank?`. If you're asking ActiveRecord if something exists given a hash that _could_ be empty, this would stop the query from executing.